pennylane/devices/qutrit_mixed/simulate.py

```diff
@@ -0,0 +1,190 @@
+    for op in circuit.operations[bool(prep) :]:
+        state = apply_operation(op, state, is_state_batched=is_state_batched, debugger=debugger)
+
+        # new state is batched if i) the old state is batched, or ii) the new op adds a batch dim
+        is_state_batched = is_state_batched or op.batch_size is not None
+
+    num_operated_wires = len(circuit.op_wires)
+    for i in range(len(circuit.wires) - num_operated_wires):
+        # If any measured wires are not operated on, we pad the density matrix with zeros.
+        # We know they belong at the end because the circuit is in standard wire-order
+        # Since it is a dm, we must pad it with 0s on the last row and last column
+        current_axis = num_operated_wires + i + is_state_batched
+        state = qml.math.stack(
+            ([state] + [qml.math.zeros_like(state)] * (QUDIT_DIM - 1)), axis=current_axis
+        )
+        state = qml.math.stack(([state] + [qml.math.zeros_like(state)] * (QUDIT_DIM - 1)), axis=-1)
+
+    return state, is_state_batched
+
+
+def measure_final_state(circuit, state, is_state_batched, rng=None, prng_key=None) -> Result:
+    """
+    Perform the measurements required by the circuit on the provided state.
+
+    This is an internal function that will be called by ``default.qutrit.mixed``.
+
+    Args:
+        circuit (.QuantumScript): The single circuit to simulate
+        state (TensorLike): The state to perform measurement on
+        is_state_batched (bool): Whether the state has a batch dimension or not.
+        rng (Union[None, int, array_like[int], SeedSequence, BitGenerator, Generator]): A
+            seed-like parameter matching that of ``seed`` for ``numpy.random.default_rng``.
+            If no value is provided, a default RNG will be used.
+        prng_key (Optional[jax.random.PRNGKey]): An optional ``jax.random.PRNGKey``. This is
+            the key to the JAX pseudo random number generator. Only for simulation using JAX.
+            If None, the default ``sample_state`` function and a ``numpy.random.default_rng``
+            will be for sampling.
+
+    Returns:
+        Tuple[TensorLike]: The measurement results
+    """
+
+    circuit = circuit.map_to_standard_wires()
+
+    if not circuit.shots:
+        # analytic case
+
+        if len(circuit.measurements) == 1:
+            return measure(circuit.measurements[0], state, is_state_batched)
+
+        return tuple(measure(mp, state, is_state_batched) for mp in circuit.measurements)
+
+    # finite-shot case
+    rng = default_rng(rng)
+    results = tuple(
+        measure_with_samples(
+            mp,
+            state,
+            shots=circuit.shots,
+            is_state_batched=is_state_batched,
+            rng=rng,
+            prng_key=prng_key,
+        )
+        for mp in circuit.measurements
+    )
+
+    if len(circuit.measurements) == 1:
+        return results[0]
+    if circuit.shots.has_partitioned_shots:
```

Comment:
There are some differences in the logic here from the qubit version, what is the reason for that?